cmd/storer2elastic/main.go

```diff
@@ -0,0 +1,283 @@
+	}
+	if ctx.IsSet(timeoutFlag.Name) {
+		configuration.General.Timeout = ctx.GlobalInt(timeoutFlag.Name)
+	}
+	if ctx.IsSet(nodeConfigFilePathFlag.Name) {
+		configuration.General.NodeConfigFilePath = ctx.GlobalString(nodeConfigFilePathFlag.Name)
+	}
+
+	err = core.LoadTomlFile(&nodeConfig, configuration.General.NodeConfigFilePath)
+	if err != nil {
+		return err
+	}
+
+	marshalizer, err = marshalFactory.NewMarshalizer(nodeConfig.Marshalizer.Type)
+	if err != nil {
+		return err
+	}
+	hasher, err = hasherFactory.NewHasher(nodeConfig.Hasher.Type)
+	if err != nil {
+		return err
+	}
+	addressPubKeyConverter, err = stateFactory.NewPubkeyConverter(nodeConfig.AddressPubkeyConverter)
+	if err != nil {
+		return err
+	}
+	validatorPubKeyConverter, err = stateFactory.NewPubkeyConverter(nodeConfig.ValidatorPubkeyConverter)
+	if err != nil {
+		return err
+	}
+
+	elasticConnectorFactoryArgs := elastic.ConnectorFactoryArgs{
+		ElasticConfig:            configuration.ElasticSearch,
+		Marshalizer:              marshalizer,
+		Hasher:                   hasher,
+		ValidatorPubKeyConverter: validatorPubKeyConverter,
+		AddressPubKeyConverter:   addressPubKeyConverter,
+	}
+
+	elasticConnectorFactory, err := elastic.NewConnectorFactory(elasticConnectorFactoryArgs)
+	if err != nil {
+		return err
+	}
+
+	elasticIndexer, err := elasticConnectorFactory.Create()
+	if err != nil {
+		return fmt.Errorf("error connecting to elastic: %w", err)
+	}
+
+	// TODO: maybe use custom configs from node config instead of a general configuration
+	generalDBConfig := config.DBConfig{
+		Type:              string(storageUnit.LvlDBSerial),
+		BatchDelaySeconds: 2,
+		MaxBatchSize:      30000,
+		MaxOpenFiles:      20,
+	}
+
+	persisterFactory := factory.NewPersisterFactory(nodeConfigPackage.DBConfig(generalDBConfig))
+	dbReaderArgs := databasereader.Args{
+		DirectoryReader:   factory.NewDirectoryReader(),
+		Marshalizer:       marshalizer,
+		PersisterFactory:  persisterFactory,
+		DbPathWithChainID: configuration.General.DBPathWithChainID,
+	}
+	dbReader, err := databasereader.New(dbReaderArgs)
+	if err != nil {
+		return err
+	}
+
+	genesisNodesConfig, err := sharding.NewNodesSetup(
+		flagsValues.nodesSetupFilePath,
+		addressPubKeyConverter,
+		validatorPubKeyConverter,
+	)
+	if err != nil {
+		return err
+	}
+
+	dataIndexerArgs := dataprocessor.Args{
+		ElasticIndexer:           elasticIndexer,
+		DatabaseReader:           dbReader,
+		ShardCoordinator:         shardCoordinator,
+		Marshalizer:              marshalizer,
+		Hasher:                   hasher,
+		GenesisNodesSetup:        genesisNodesConfig,
+		Uint64ByteSliceConverter: uint64ByteSliceConverter,
+	}
+	dataIndexer, err := dataprocessor.New(dataIndexerArgs)
+	if err != nil {
+		return err
+	}
+
+	err = dataIndexer.Index(configuration.General.Timeout)
+	if err != nil {
+		return err
+	} else {
+		log.Info("indexing finished")
+	}
+
+	waitForUserToTerminateApp()
```

Comment:
This does not work as expected. dataIndexer.Index is blocking and after is finishes, will wait for the user's input to close the application instead of turning it off by itself.

Reply:
removed. now when the indexing finishes (or fails) the application will stop